src/repr/src/adt/interval.rs

```diff
@@ -268,6 +279,7 @@ impl Interval {
     {
         T::from(self.years() * 60 * 60 * 24) * T::from(365.25)
             + T::from(self.months() * 60 * 60 * 24 * 30)
+            + T::from(self.days * 60 * 60 * 24)
```

Comment:
This can overflow is days is i32::MAX.

Reply:
I replaced all the potentially overflowing arithmetic with float arithmetic, which should return +inf/-inf instead of overflowing. I also added tests that use this function with the min/max Intervals